index.js

```diff
@@ -2,60 +2,77 @@ var dir = require("node-dir");
 var path = require("path");
 var fs = require("fs");
 
-module.exports = function(folder, cb){
+module.exports = function(folder, cb) {
 
-	dir.files(folder, function(err, files) {
-		if(err){
-			cb(err);
+	var cwd = process.cwd();
+	process.chdir(folder);
+
+	// Wrap the original callback so we don't forget to change directory back
+	var cbWrapper = function () {
+		process.chdir(cwd);
+		cb.apply(null, arguments);
+	};
+
+	dir.files(process.cwd(), function(err, files) {
+		if (err) {
+			cbWrapper(err);
 		}
-		else{
-			var moduleNames = ["active"];
+		else {
+			var moduleNames = ["main"];
 			var filesByModule = {
-				active: []
+				main: []
 			};
-			for(var i=0; i<files.length; i++){
-				var file = files[i].replace(folder, "");
-
-				if(file.match(/\.js$/)){
-					var bits = file.split(path.sep);
-					if(bits[1] == "node_modules"){
-						filesByModule[bits[2]] = filesByModule[bits[2]] || [];
-						filesByModule[bits[2]].push(files[i]);
+			for (var i = 0; i < files.length; i++) {
+				var file = files[i];
+				var fileParts = file.split(path.sep);
+				var lastIdx = fileParts.lastIndexOf("node_modules");
+
+				if (file.match(/\.js$/)) {
+					if (lastIdx >= 0) {
```

Comment:
nested node_modules, e.g., for file `node_modules/foo/node_modules/bar/node_modules/baz/index.js`

before, module name was detected as "foo" - now, detected as "baz"